langgraph/prebuilt/tool_executor.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Sequence, Union
+
+from langchain_core.load.serializable import Serializable
+from langchain_core.runnables import RunnableBinding, RunnableLambda
+from langchain_core.tools import BaseTool
+
+INVALID_TOOL_MSG_TEMPLATE = (
+    "{requested_tool_name} is not a valid tool, "
+    "try one of [{available_tool_names_str}]."
+)
+
+
+class ToolInvocationInterface:
+    """Interface for invoking a tool"""
+
+    tool: str
+    tool_input: Union[str, dict]
+
+
+class ToolInvocation(Serializable):
+    """Information about how to invoke a tool."""
+
+    tool: str
```

Comment:
Should this take a tool id for multiple tool invocation use case? (when invoking the same tool, but with different arguments in parallel)